datalake/src/main/java/com/sequenceiq/datalake/flow/delete/SdxDeleteActions.java

```diff
@@ -147,6 +152,13 @@ protected void doExecute(SdxContext context, RdsDeletionSuccessEvent payload, Ma
                 if (sdxCluster != null) {
                     metricService.incrementMetricCounter(MetricType.SDX_DELETION_FINISHED, sdxCluster);
                 }
+                // If the sdx that is deleted is detached, we need to replace CRN with the CRN of the original SDX
```

Comment:
Wouldn't it make more sense to do the swapping during the last step of the creation of the replacement? 
There could be a timeframe where someone can issue operations on the newly created but not swapped DL.
I'm asking because what if the old DL get's stuck for some reason, like someone does a repair in the background, or any other operations happens?

Reply:
why do we need to replace the CRN with original one? or other question is that why does CRN change at all during resize? second question may was answered in the past but I cannot remember

Reply:
Laszlo,
Ideally yes. If I do it there will be having two non-deleted data lakes using the CRN. As part of CB-13515, we will be updating the name at the same time. That is why picked the point where the old data lake is deleted.


Adam,
WIth resizing we will be creating a parallel SDX before we delete the older one. This makes recovery simple.
As we creating a new SDX, the CRN of the newly created SDX will be different. 
All the experiences use the CRN of the data lake to pull the updated configuration. It is critical to maintain the same CRN after resizing.


https://docs.google.com/document/d/1KbPCF6Sem5YkWIbAk5GkbG2vM_cqCzhFRGJfR10aNQg/edit?usp=sharing